snappy/firstboot.go

```diff
@@ -46,39 +47,63 @@ func wrapConfig(pkgName string, conf interface{}) ([]byte, error) {
 	return yaml.Marshal(configWrap)
 }
 
-var activeSnapByName = ActiveSnapByName
-var activeSnapsByType = ActiveSnapsByType
+var newPkgmap = newPkgmapImpl
+
+func newPkgmapImpl() (map[string]Part, error) {
+	repo := NewMetaLocalRepository()
+	all, err := repo.All()
+	if err != nil {
+		return nil, err
+	}
+
+	m := make(map[string]Part, 2*len(all))
+	for _, part := range all {
+		m[FullName(part)] = part
+		m[BareName(part)] = part
+	}
+
+	return m, nil
+}
 
 // OemConfig checks for an oem snap and if found applies the configuration
 // set there to the system
 func oemConfig() error {
-	oemSnap, err := activeSnapsByType(pkg.TypeOem)
-	if err != nil {
+	oem, err := getOem()
+	if err != nil || oem == nil {
 		return err
 	}
 
-	if len(oemSnap) < 1 {
-		return nil
-	}
-
-	snap, ok := oemSnap[0].(Configuration)
-	if !ok {
-		return ErrNoOemConfiguration
+	pkgmap, err := newPkgmap()
+	if err != nil {
+		return err
 	}
 
-	for pkgName, conf := range snap.OemConfig() {
-		configData, err := wrapConfig(pkgName, conf)
-		if err != nil {
-			return err
+	pb := progress.MakeProgressBar()
+	for _, pkgName := range oem.OEM.Software.BuiltIn {
+		part, ok := pkgmap[pkgName]
+		if !ok {
+			return errNoSnapToActivate
+		}
+		snap, ok := part.(*SnapPart)
+		if !ok {
+			return errNoSnapToActivate
 		}
+		snap.activate(false, pb)
```

Comment:
Previously there so not a blocker for this branch, but would be nice to eventually have the bool in a flags parameter or config struct, making it slightly more readable.